packages/eslint-plugin-turbo/lib/rules/no-undeclared-env-vars.ts

```diff
@@ -67,11 +99,70 @@ function normalizeCwd(
   return undefined;
 }
 
+/** for a given `package.json` file path, this will compile a Set of that package's listed dependencies */
+const packageJsonDependencies = (filePath: string): Set<string> => {
+  // get the contents of the package.json
+  const packageJsonString = readFileSync(filePath, "utf-8");
+  const packageJson = JSON.parse(packageJsonString) as Record<
+    string,
+    undefined | Record<string, string>
+  >;
+
+  return [
+    "dependencies",
+    "devDependencies",
+    "peerDependencies",
+    // intentionally not including `optionalDependencies` or `bundleDependencies` because at the time of writing they are not used for any of the frameworks we support
+  ]
+    .flatMap((key) => Object.keys(packageJson[key] ?? {}))
+    .reduce((acc, dependency) => acc.add(dependency), new Set<string>());
```

Comment:
Not required, but feels like a weird use case for reduce. You could just create the Set outside of the return, and then:

```js
const deps = new Set()

[
    "dependencies",
    "devDependencies",
    "peerDependencies",
].flatMap((key) => Object.keys(packageJson[key] ?? {})).forEach(deps.add)

return deps
```

Reply:
I wish we could inline `deps.add` but unfortunately we have an "unbounded-method" rule on (well, fortunate to have the rule on, unfortunate that there's a JavaScript footgun in doing this like you're subscribing).

So it'd have to be

```diff
-  return [
+  const deps = new Set<string>();
+
+  [
     "dependencies",
     "devDependencies",
     "peerDependencies",
     // intentionally not including `optionalDependencies` or `bundleDependencies` because at the time of writing they are not used for any of the frameworks we support
   ]
     .flatMap((key) => Object.keys(packageJson[key] ?? {}))
-    .reduce((acc, dependency) => acc.add(dependency), new Set<string>());
+    .forEach((dependency) => deps.add(dependency));
+
+  return deps;
```

Would you like it to be like that?  Seems normal to use reduce because we're building up the Set with each iteration's new data, but I'm cool with whatever! :)

Reply:
Ah yea - one of my favorite JS patterns that TS hates ha, but it makes sense why. 

Nah it's fine to leave as is!